src/main/scala/io/scalac/tezos/translator/service/LibraryService.scala

```diff
@@ -6,20 +6,22 @@ import io.scalac.tezos.translator.repository.LibraryRepository
 import io.scalac.tezos.translator.repository.dto.LibraryEntryDbDto
 import slick.jdbc.MySQLProfile.api._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
 
 class LibraryService(repository: LibraryRepository, db: Database) {
 
   def addNew(entry: LibraryEntry): Future[Int] =
     db.run(repository.add(LibraryEntryDbDto.fromDomain(entry)))
 
-  def getAccepted(count: Int)(implicit ec: ExecutionContext): Future[Seq[LibraryEntry]] =
-    getAll(count, Some(Accepted))
+  def getAccepted(limit: Int): Future[Seq[LibraryEntry]] =
+    getAll(limit, Some(Accepted))
 
-  def getAll(count: Int, statusFilter: Option[Status] = None)(implicit ec: ExecutionContext): Future[Seq[LibraryEntry]] =
+  def getAll(limit: Int = Integer.MAX_VALUE, statusFilter: Option[Status] = None): Future[Seq[LibraryEntry]] = {
+    implicit val executor: ExecutionContextExecutor = ExecutionContext.global
```

Comment:
Please do not define `ExecutionContext` inside of a function. it should be passed implicitly to a function or to a class (`class LibraryService(repository: LibraryRepository, db: Database)(implicit ex: ExecutionContext)`).
Such approach allows to provide a specific execution context if needed. In this application `ec` is created in `Boot.scala` and shoudl be propagated elsewhere.
As a rule of thumb`global` ec shouldnt be used inside an application.

Reply:
Right, I changed it because it doesn't seem right to use actor system's dispatcher to communicate with database (especially if not from within an actor), but you're right that global could be even worse.

Reply:
You are right. The thread pools should be revised in this project. I think there is a blocking one for sending emails... so it should be reused for db connections. I will add a task for that